packages/components/icon/src/icon.vue

```diff
@@ -22,16 +22,11 @@ export default defineComponent({
     const ns = useNamespace('icon')
 
     const style = computed<CSSProperties>(() => {
-      if (!props.size && !props.color) {
-        return {}
-      }
-      let size = props.size
-      if (isNumber(size) || (isString(size) && !size.endsWith('px'))) {
-        size = `${size}px`
-      }
+      if (!props.size && !props.color) return {}
+
       return {
-        ...(props.size ? { fontSize: size } : {}),
-        ...(props.color ? { '--color': props.color } : {}),
+        fontSize: !isUndefined(props.size) ? addUnit(props.size) : undefined,
```

Comment:
**Suggestion:** For me here, we might not need this `negation` for `props.size` because we can swap the branches in your code to get rid of the `negation` operator here.

```ts
isUndefined(props.size) ? undefined : addUnit(props.size)
```

Reply:
The whole idea is that we need to avoid unnecessary operations which in this case the `negation`.

Reply:
Updated.